source/base/parameter_handler.cc

```diff
@@ -2311,13 +2319,391 @@ ParameterHandler::print_parameters (std::ostream     &out,
     }
 
   // dive recursively into the subsections
-  print_parameters_section (out, style, 0);
+  recursively_print_parameters (std::vector<std::string>(), // start at the top level
+                                style,
+                                0,
+                                out);
 
   return out;
 }
 
 
 
+void
+ParameterHandler::recursively_print_parameters (const std::vector<std::string> &target_subsection_path,
+                                                const OutputStyle               style,
+                                                const unsigned int              indent_level,
+                                                std::ostream                   &out) const
+{
+  AssertThrow (out, ExcIO());
+
+  Assert ((style != XML) && (style != JSON),
+          ExcMessage ("This function should not be necessary for XML or JSON output."));
```

Comment:
This wasn't really intended as a user-facing  error message. The function in question here is `private`, and no other function in this class should call it with these `style` arguments. I'll turn this into an `ExcInternalError` and move the message into a comment.